src/tools/decomp.py

```diff
@@ -53,135 +100,388 @@
+        raise Error("local box size in direction 1 (=%6f) is smaller than interaction range (cutoff + skin = %6f).\n \
+                 hint: number of CPUs maybe too high or is prime, perhaps you could also try with Halfcells." % (iy, rc_skin))
+    iz = box_size[2] / (rc_skin * node_grid[2])
+    if iz < 1:
+        raise Error("local box size in direction 2 (=%6f) is smaller than interaction range (cutoff + skin = %6f).\n \
+                 hint: number of CPUs maybe too high or is prime, perhaps you could also try with Halfcells." % (iz, rc_skin))
+
+    return Int3D(ix, iy, iz)
+
+
+def nodeGrid_OLD(n):
+    ijkmax = 3 * n * n + 1
+    d1 = 1
+    d2 = 1
+    d3 = 1
+    for i in xrange(1, n + 1):
+        for j in xrange(i, n + 1):
+            for k in xrange(j, n + 1):
+                if (i * j * k == n) and (i * i + j * j + k * k < ijkmax):
+                    d1 = k
+                    d2 = j
+                    d3 = i
+                    ijkmax = i * i + j * j + k * k
+    return Int3D(d1, d2, d3)
+
+
+def cherrypickTotalProcs(box_size, rc, skin, MnN, CpN, percTol=0.2, eh_size=0, ratioMS=0, idealGas=0, slabMSDims=[0, 0, 0]):
+    indMax = max(box_size)
+    indMin = min(box_size)
+    L_l = box_size[indMax]
+    L_s = box_size[indMin]
+    L_lHR = 2. * eh_size
+    L_lLR = L_l - L_lHR
+    SymFactor = int((L_l + L_lHR * (ratioMS**(1. / 3.) - 1.)) / (L_s * (ratioMS**(1. / 3.) - 1.)))
+    nX = [0] * MnN
+    nY = [0] * MnN
+    nZ = [0] * MnN
+    procArray = range(1 * CpN, (MnN + 1) * CpN, *CpN)
+    print "HeSpaDDA message: Your search array in terms of total number of processors is:", procArray
+    for i in xrange(1, MnN + 1):
+        nX[i - 1], nY[i - 1], nZ[i - 1] = nodeGrid(box_size, rc, skin, procArray[i - 1], eh_size, ratioMS, idealGas, slabMSDims)
+    print "HeSpaDDA message: For your Information, we are tackling the following number of processors"
+    print "HeSpaDDA message: Processors in the x-axis are:", nX, "\nProcessors in the y-axis are:", nY, "\nProcessors in the z-axis are:", nZ
+    j = []
+    for i in xrange(0, MnN):
+        if (indMax == 0 and indMin == 1) or (indMax == 0 and indMin == 2):
+            if nX[i] > (SymFactor - percTol * SymFactor) * nY[i] and nX[i] < (SymFactor + percTol * SymFactor) * nY[i]:
+                j.append(i)
+        elif (indMax == 1 and indMin == 0) or (indMax == 1 and indMin == 2):
+            if nY[i] > (SymFactor - percTol * SymFactor) * nX[i] and nY[i] < (SymFactor + percTol * SymFactor) * nX[i]:
+                j.append(i)
+        elif (indMax == 2 and indMin == 0) or (indMax == 2 and indMin == 1):
+            if nZ[i] > (SymFactor - percTol * SymFactor) * nX[i] and nZ[i] < (SymFactor + percTol * SymFactor) * nX[i]:
+                j.append(i)
+    print "HeSpaDDA message: There are a couple of total number of processors that satisfy your requirements: ", [(j[i] + 1) * CpN for i in xrange(len(j))]
+    return [(j[i] + 1) * CpN for i in xrange(len(j))]
```

Comment:
if you iterate over the elements of ```j``` then simple
```py
return [(v+1)*CpN for v in j]
```

Reply:
true and more sophisticated

Reply:
It was converted from the original version using numpy...now without numpy